src/utils/format_utils.hxx

```diff
@@ -163,4 +164,17 @@ inline string ipv4_to_string(uint32_t ip)
     return ipv4_to_string((uint8_t*)&ip);
 }
 
+/// Populates a character array with a C string. Copies the C string,
+/// appropriately truncating if it is too long and filling the remaining space
+/// with zeroes. Ensures that at least one null terminator character is
```

Comment:
The comment of "filling the remaining space with zeros" does not seem to be implemented here. Perhaps adding:
```
bzero(dst, N);
```
would satisfy that?

Reply:
Actually it does fill with null: https://pubs.opengroup.org/onlinepubs/009695399/functions/strncpy.html

>If the array pointed to by s2 is a string that is shorter than n bytes, null bytes shall be appended to the copy in the array  pointed to by s1, until n bytes in all are written.

Reply:
Ok, so based on that it should be zeroing the array automatically. We can consider this a non-issue in that case.